e2e/shared-mode-smoke-test.sh

```
@@ -381,6 +550,50 @@ else
+# Peer tenant must not see the completed multipart object.
+resp=$(http GET /v1/fs/big.bin "$A_KEY")
+check_eq "A cannot read B's multipart /big.bin" "404" "${resp%%|*}"
+# Upload session ownership: A cannot complete B's in-flight upload_id.
+jq -n --arg path "/iso-upload.bin" --argjson total_size 10485760 --arg checksums "$CHECKSUMS" \
+  '{path:$path,total_size:$total_size,part_checksums:($checksums|split(","))}' > "$TMP_DIR/init-iso.json"
+code=$(curl -sS -o "$TMP_DIR/plan-iso.json" -w "%{http_code}" -X POST \
+  -H "Authorization: Bearer $B_KEY" -H "Content-Type: application/json" \
+  --data-binary "@$TMP_DIR/init-iso.json" "http://127.0.0.1:$SERVER_PORT/v1/uploads/initiate")
+check_eq "B isolation multipart initiate returns 202" "202" "$code"
+ISO_UPLOAD_ID=$(jq -r '.upload_id // empty' "$TMP_DIR/plan-iso.json")
+[ -n "$ISO_UPLOAD_ID" ] && pass "B isolation upload_id issued" || fail "B isolation upload_id empty"
```

Shellcheck (0.11.0)

[info] Note that A && B || C is not if-then-else. C may run when A is true.

(SC2015)